core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -50,9 +54,47 @@
   private Deletes() {
   }
 
-  public static <T> CloseableIterable<T> positionFilter(CloseableIterable<T> rows, Function<T, Long> rowToPosition,
-                                                        CloseableIterable<Long> posDeletes) {
-    return new PositionDeleteFilter<>(rows, rowToPosition, posDeletes);
+  public static <T> CloseableIterable<T> equalitySetFilter(CloseableIterable<T> rows,
+                                                           Function<T, StructLike> rowToDeleteKey,
+                                                           Types.StructType eqType,
+                                                           CloseableIterable<StructLike> eqDeletes) {
+    try (CloseableIterable<StructLike> deletes = eqDeletes) {
+      CloseableIterator<StructLike> eqDeleteIterator = deletes.iterator();
+      if (eqDeleteIterator.hasNext()) {
+        StructLikeSet deleteSet = StructLikeSet.create(eqType);
```

Comment:
Maybe this set can be reused? A set gets bigger and bigger in merging?
So can we have a method like `addAll(CloseableIterable<StructLike>)` in `StructLikeSet`?

Reply:
I refactored this so that the filter method accepts a set. That will make them reusable.